### ISSUE TITLE
[Research] Add FACE Few-Shot Adapter for Per-User EEG Emotion Personalization

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#20]
## What

Implement the **FACE (Few-shot Adapter with Cross-view Fusion)** framework (arXiv Mar 2025) to enable **per-user personalization** of emotion classification with only ~10 labeled samples per emotion class.

## Why

- Current system uses **population-average thresholds** — no personalization
- Known issue: within-subject accuracy is **26 points higher** than cross-subject (90.97% vs 64.82%)
- FACE closes this gap using meta-learning adapters that personalize in ~1-2 minutes of user calibration
- First work to apply adapter mechanisms to cross-domain EEG emotion recognition
- Outperforms SOTA on SEED, SEED-IV, and SEED-V benchmarks
- JMIR 2024 study confirms: personalized models achieve **95.06%** vs generalized **66.95%** on consumer wearables

### Step 1: Implement FACE adapter module
```python
# ml/models/face_adapter.py
class FACEAdapter(nn.Module):
    """Few-shot adapter with cross-view fusion.
    
    Components:
    1. Cross-view fusion: dynamically integrates global brain connectivity 
       with localized patterns via subject-specific fusion weights
    2. Few-shot adapter: meta-learning for rapid adaptation to unseen subjects
    """
    def __init__(self, base_feature_dim=41, num_emotions=6):
        # Adapter layers (small, trainable per-user)
        # Cross-view fusion weights (subject-specific)
        pass
    
    def personalize(self, calibration_samples, calibration_labels):
        """Fine-tune adapter on ~10 samples per class (60 total for 6-class)."""
        pass
```

### Step 2: Create calibration UI flow
- Add "Calibration Mode" button to EEG recording page
- Show 6 short emotion-inducing stimuli (video clips or images)
- Record 10-15 seconds of EEG per stimulus
- User rates their felt emotion (ground truth label)
- Total time: ~2 minutes

### Step 3: Meta-learning training loop
- Pre-train base model on SEED/DEAP data using MAML or Reptile
- At calibration time, run 5-10 gradient steps on user's samples
- Store adapter weights per user in database

### Step 4: Wire into inference
- After calibration: use personalized adapter for all predictions
- Before calibration: fall back to population-average heuristics (current behavior)
- Store adapter weights in user profile (PostgreSQL JSONB or file)

### Files to Modify
- `ml/models/face_adapter.py` (NEW) — FACE adapter implementation
- `ml/models/emotion_classifier.py` — Add personalized path
- `ml/training/train_face_meta.py` (NEW) — Meta-learning pre-training
- `ml/api/routes.py` — Add `/calibration/personalize` endpoint
- `client/src/pages/eeg-monitor.tsx` — Add calibration UI
- `shared/schema.ts` — Add user_adapter_weights field

### Expected Impact
- Per-user emotion accuracy: **74% → 85-92%** after 2-min calibration
- Closes the 26-point within-subject vs cross-subject gap

## Complexity
MEDIUM-HIGH — meta-learning training, calibration UI, per-user weight storage

## References
- FACE paper: https://arxiv.org/abs/2503.18998
- JMIR 2024 personalized vs generalized: https://ai.jmir.org/2024/1/e52171
- EmoAdapt (4-ch self-supervised): https://www.mdpi.com/2227-7390/13/10/1608
